orbit/models/ktrlite.py

```diff
@@ -271,10 +282,13 @@ def _set_kernel_matrix(self, df):
             self.knots_tp_level = (1 + knots_idx_level) / self.num_of_observations
             self._level_knot_dates = df[self.date_col].values[knots_idx_level]
         else:
+            # to exclude dates which are not within training period
             self._level_knot_dates = pd.to_datetime([
-                x for x in self._level_knot_dates if 
-                (x <= df[self.date_col].max()) and (x >= df[self.date_col].min())
+                x for x in self._level_knot_dates if
+                (x <= df[self.date_col].values[-1]) and (x >= df[self.date_col].values[0])
```

Comment:
this is assuming that `df` is sorted by `date_col`, though this should be a requirement for our model.

Reply:
just curious if there is any special consideration in this change?

Reply:
i have some impression that is what we check by default in the see` _validate_training_df` under base template.